src/lua/uri.c

```diff
@@ -161,12 +161,53 @@ uri_create_from_lua_table(struct lua_State *L, int idx, struct uri *uri)
 	lua_pop(L, 1);
 	if (rc != 0)
 		return rc;
+
+	bool has_login = false;
+	lua_pushstring(L, "login");
+	lua_rawget(L, idx);
```

Comment:
Nit: You can use `lua_getfield` helper here.

Reply:
Thanks, used.